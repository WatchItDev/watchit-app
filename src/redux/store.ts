import { type Store } from 'redux';
import { configureStore } from '@reduxjs/toolkit';
import rootReducer from '@src/redux/reducer';

export const createStore = (initialState = {}): Store => {
  return configureStore({
    reducer: rootReducer,
    devTools: true,
    preloadedState: initialState,
    middleware: (getDefaultMiddleware) =>
      getDefaultMiddleware({
        serializableCheck: false,
      }),
  });
};

export type RootState = ReturnType<typeof store.getState>;

export const store = createStore();
