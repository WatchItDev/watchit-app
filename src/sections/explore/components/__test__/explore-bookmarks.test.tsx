import { render, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import { store } from '@redux/store.ts';
import { MemoryRouter } from 'react-router-dom';
import { MockedProvider } from '@apollo/client/testing';

vi.mock('@src/workers/backgroundTaskWorker?worker', () => {
  return {
    default: class {
      postMessage() {}
      terminate() {}
      addEventListener() {}
      removeEventListener() {}
    },
  };
});

vi.mock('@src/hooks/use-responsive.ts', () => ({
  useResponsive: () => true,
  useWidth: () => 1024,
}));

const renderWithProviders = async () => {
  const { ExploreBookmarks } = await import('../explore-bookmarks');
  return render(
    <Provider store={store}>
      <MockedProvider>
        <MemoryRouter
          future={{ v7_startTransition: true, v7_relativeSplatPath: true }}
        >
          <ExploreBookmarks />
        </MemoryRouter>
      </MockedProvider>
    </Provider>,
  );
};

vi.mock('@src/hooks/use-bookmark.ts', () => ({
  useBookmarks: () => ({
    data: [
      {
        id: '1',
        description: 'Bookmark 1 description',
        isHidden: false,
        likeCount: 100,
      },
      {
        id: '2',
        description: 'Bookmark 2 description',
        isHidden: false,
        likeCount: 200,
      },
      {
        id: '3',
        description: 'Bookmark 3 description',
        isHidden: true,
        likeCount: 150,
      },
    ],
  }),
}));

describe('Testing in the <ExploreBookmarks/> component', () => {
  it('should match snapshot', async () => {
    const { container } = await renderWithProviders();
    expect(container).toMatchSnapshot();
  });

  it('should reverse the order of bookmarkPublications', async () => {
    await renderWithProviders();
    expect(screen.getByText('Bookmark 2 description')).toBeInTheDocument();
    expect(screen.getByText('Bookmark 1 description')).toBeInTheDocument();
    expect(
      screen.queryByText('Bookmark 3 description'),
    ).not.toBeInTheDocument();
  });

  it('should render the ExploreBookmarks component', async () => {
    await renderWithProviders();
    expect(screen.getByText('Bookmarks')).toBeInTheDocument();
    expect(screen.getByText('Bookmark 1 description')).toBeInTheDocument();
    expect(screen.getByText('Bookmark 2 description')).toBeInTheDocument();
  });

  it('should not render hidden bookmarks', async () => {
    await renderWithProviders();
    expect(
      screen.queryByText('Bookmark 3 description'),
    ).not.toBeInTheDocument();
  });
  it('should render bookmarks with unique ids', async () => {
    await renderWithProviders();
    const bookmarks = screen.getAllByText(/Bookmark \d description/);
    const uniqueIds = new Set(
      bookmarks.map((bookmark) => bookmark.textContent),
    );
    expect(uniqueIds.size).toBe(bookmarks.length);
  });

  it('should render bookmarks with correct likeCounts', async () => {
    await renderWithProviders();
    expect(screen.getByText(/100/)).toBeInTheDocument();
    expect(screen.getByText(/200/)).toBeInTheDocument();
    expect(screen.queryByText(/150/)).not.toBeInTheDocument();
  });
});
