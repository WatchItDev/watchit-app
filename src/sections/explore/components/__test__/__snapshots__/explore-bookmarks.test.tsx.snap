// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[` Testing in the <explore-bookmarks/> component  >  should match snapshot  1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-bhp9pd-MuiPaper-root-MuiCard-root"
  >
    <div
      class="MuiCardHeader-root css-1pkklwm-MuiCardHeader-root"
    >
      <div
        class="MuiCardHeader-content css-1qbkelo-MuiCardHeader-content"
      >
        <span
          class="MuiTypography-root MuiTypography-h5 MuiCardHeader-title css-1qvr50w-MuiTypography-root"
        >
          Bookmarks
        </span>
      </div>
      <div
        class="MuiCardHeader-action css-sgoict-MuiCardHeader-action"
      >
        <button
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
          data-testid="leftButton"
          tabindex="0"
          type="button"
        >
          <svg
            class="tabler-icon tabler-icon-chevron-left "
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M15 6l-6 6l6 6"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
        <button
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
          data-testid="rightButton"
          tabindex="0"
          type="button"
        >
          <svg
            class="tabler-icon tabler-icon-chevron-right "
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M9 6l6 6l-6 6"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
    <div
      class="MuiCardContent-root css-fd8ij7-MuiCardContent-root"
    >
      <div
        class="MuiBox-root css-1v61vtn"
      >
        <div
          class="slick-slider slick-initialized"
        >
          <div
            class="slick-list"
            style="height: 0px;"
          >
            <div
              class="slick-track"
              style="opacity: 1; transform: translate3d(0px, 0px, 0px);"
            >
              <div
                aria-hidden="false"
                class="slick-slide slick-active slick-current"
                data-index="0"
                style="outline: none; width: 0px;"
                tabindex="-1"
              >
                <div>
                  <div
                    class="MuiBox-root css-0"
                  >
                    <div
                      class="MuiBox-root css-k008qs"
                    >
                      <div
                        class="MuiBox-root css-rmtq2p"
                      >
                        <div
                          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-11rpxa7-MuiPaper-root"
                        >
                          <span
                            class="component-image MuiBox-root css-1jl2tef"
                            style="border-radius: 10px;"
                          >
                            <span
                              class="component-image-wrapper lazy-load-image-background blur"
                              style="background-image: url(/assets/placeholder.svg); background-size: 100% 100%; color: transparent; display: inline-block;"
                            >
                              <img
                                class="MuiBox-root css-19t2nho"
                                src=""
                              />
                            </span>
                          </span>
                          <div
                            class="MuiBox-root css-1ue9rvn"
                          >
                            <div
                              class="MuiBox-root css-1lekzkb"
                            >
                              <div
                                class="MuiStack-root css-1nnjrde-MuiStack-root"
                              >
                                <svg
                                  class="tabler-icon tabler-icon-heart-filled "
                                  fill="#F2F3F5"
                                  height="16"
                                  stroke="none"
                                  style="margin-bottom: 2px;"
                                  viewBox="0 0 24 24"
                                  width="16"
                                  xmlns="http://www.w3.org/2000/svg"
                                >
                                  <path
                                    d="M6.979 3.074a6 6 0 0 1 4.988 1.425l.037 .033l.034 -.03a6 6 0 0 1 4.733 -1.44l.246 .036a6 6 0 0 1 3.364 10.008l-.18 .185l-.048 .041l-7.45 7.379a1 1 0 0 1 -1.313 .082l-.094 -.082l-7.493 -7.422a6 6 0 0 1 3.176 -10.215z"
                                  />
                                </svg>
                                <p
                                  class="MuiTypography-root MuiTypography-body2 css-1571v2w-MuiTypography-root"
                                >
                                  100
                                </p>
                              </div>
                            </div>
                          </div>
                          <div
                            class="MuiCardContent-root css-gbn0xv-MuiCardContent-root"
                          >
                            <h6
                              class="MuiTypography-root MuiTypography-h6 MuiTypography-noWrap css-z9ruai-MuiTypography-root"
                            />
                            <div
                              class="MuiStack-root css-kdc3n8-MuiStack-root"
                            >
                              <p
                                class="MuiTypography-root MuiTypography-body2 css-c2tnzq-MuiTypography-root"
                              >
                                Bookmark 1 description
                              </p>
                              <p
                                class="MuiTypography-root MuiTypography-body2 MuiTypography-noWrap css-176gd80-MuiTypography-root"
                              />
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                    <div
                      class="MuiBox-root css-k008qs"
                    >
                      <div
                        class="MuiBox-root css-rmtq2p"
                      >
                        <div
                          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-11rpxa7-MuiPaper-root"
                        >
                          <span
                            class="component-image MuiBox-root css-1jl2tef"
                            style="border-radius: 10px;"
                          >
                            <span
                              class="component-image-wrapper lazy-load-image-background blur"
                              style="background-image: url(/assets/placeholder.svg); background-size: 100% 100%; color: transparent; display: inline-block;"
                            >
                              <img
                                class="MuiBox-root css-19t2nho"
                                src=""
                              />
                            </span>
                          </span>
                          <div
                            class="MuiBox-root css-1ue9rvn"
                          >
                            <div
                              class="MuiBox-root css-1lekzkb"
                            >
                              <div
                                class="MuiStack-root css-1nnjrde-MuiStack-root"
                              >
                                <svg
                                  class="tabler-icon tabler-icon-heart-filled "
                                  fill="#F2F3F5"
                                  height="16"
                                  stroke="none"
                                  style="margin-bottom: 2px;"
                                  viewBox="0 0 24 24"
                                  width="16"
                                  xmlns="http://www.w3.org/2000/svg"
                                >
                                  <path
                                    d="M6.979 3.074a6 6 0 0 1 4.988 1.425l.037 .033l.034 -.03a6 6 0 0 1 4.733 -1.44l.246 .036a6 6 0 0 1 3.364 10.008l-.18 .185l-.048 .041l-7.45 7.379a1 1 0 0 1 -1.313 .082l-.094 -.082l-7.493 -7.422a6 6 0 0 1 3.176 -10.215z"
                                  />
                                </svg>
                                <p
                                  class="MuiTypography-root MuiTypography-body2 css-1571v2w-MuiTypography-root"
                                >
                                  200
                                </p>
                              </div>
                            </div>
                          </div>
                          <div
                            class="MuiCardContent-root css-gbn0xv-MuiCardContent-root"
                          >
                            <h6
                              class="MuiTypography-root MuiTypography-h6 MuiTypography-noWrap css-z9ruai-MuiTypography-root"
                            />
                            <div
                              class="MuiStack-root css-kdc3n8-MuiStack-root"
                            >
                              <p
                                class="MuiTypography-root MuiTypography-body2 css-c2tnzq-MuiTypography-root"
                              >
                                Bookmark 2 description
                              </p>
                              <p
                                class="MuiTypography-root MuiTypography-body2 MuiTypography-noWrap css-176gd80-MuiTypography-root"
                              />
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Testing in the <ExploreBookmarks/> component > should match snapshot 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-bhp9pd-MuiPaper-root-MuiCard-root"
  >
    <div
      class="MuiCardHeader-root css-1pkklwm-MuiCardHeader-root"
    >
      <div
        class="MuiCardHeader-content css-1qbkelo-MuiCardHeader-content"
      >
        <span
          class="MuiTypography-root MuiTypography-h5 MuiCardHeader-title css-1qvr50w-MuiTypography-root"
        >
          Bookmarks
        </span>
      </div>
      <div
        class="MuiCardHeader-action css-sgoict-MuiCardHeader-action"
      >
        <button
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
          data-testid="leftButton"
          tabindex="0"
          type="button"
        >
          <svg
            class="tabler-icon tabler-icon-chevron-left "
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M15 6l-6 6l6 6"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
        <button
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
          data-testid="rightButton"
          tabindex="0"
          type="button"
        >
          <svg
            class="tabler-icon tabler-icon-chevron-right "
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M9 6l6 6l-6 6"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
    <div
      class="MuiCardContent-root css-fd8ij7-MuiCardContent-root"
    >
      <div
        class="MuiBox-root css-1v61vtn"
      >
        <div
          class="slick-slider slick-initialized"
        >
          <div
            class="slick-list"
            style="height: 0px;"
          >
            <div
              class="slick-track"
              style="opacity: 1; transform: translate3d(0px, 0px, 0px);"
            >
              <div
                aria-hidden="false"
                class="slick-slide slick-active slick-current"
                data-index="0"
                style="outline: none; width: 0px;"
                tabindex="-1"
              >
                <div>
                  <div
                    class="MuiBox-root css-0"
                  >
                    <div
                      class="MuiBox-root css-k008qs"
                    >
                      <div
                        class="MuiBox-root css-rmtq2p"
                      >
                        <div
                          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-11rpxa7-MuiPaper-root"
                        >
                          <span
                            class="component-image MuiBox-root css-1jl2tef"
                            style="border-radius: 10px;"
                          >
                            <span
                              class="component-image-wrapper lazy-load-image-background blur"
                              style="background-image: url(/assets/placeholder.svg); background-size: 100% 100%; color: transparent; display: inline-block;"
                            >
                              <img
                                class="MuiBox-root css-19t2nho"
                                src=""
                              />
                            </span>
                          </span>
                          <div
                            class="MuiBox-root css-1ue9rvn"
                          >
                            <div
                              class="MuiBox-root css-1lekzkb"
                            >
                              <div
                                class="MuiStack-root css-1nnjrde-MuiStack-root"
                              >
                                <svg
                                  class="tabler-icon tabler-icon-heart-filled "
                                  fill="#F2F3F5"
                                  height="16"
                                  stroke="none"
                                  style="margin-bottom: 2px;"
                                  viewBox="0 0 24 24"
                                  width="16"
                                  xmlns="http://www.w3.org/2000/svg"
                                >
                                  <path
                                    d="M6.979 3.074a6 6 0 0 1 4.988 1.425l.037 .033l.034 -.03a6 6 0 0 1 4.733 -1.44l.246 .036a6 6 0 0 1 3.364 10.008l-.18 .185l-.048 .041l-7.45 7.379a1 1 0 0 1 -1.313 .082l-.094 -.082l-7.493 -7.422a6 6 0 0 1 3.176 -10.215z"
                                  />
                                </svg>
                                <p
                                  class="MuiTypography-root MuiTypography-body2 css-1571v2w-MuiTypography-root"
                                >
                                  100
                                </p>
                              </div>
                            </div>
                          </div>
                          <div
                            class="MuiCardContent-root css-gbn0xv-MuiCardContent-root"
                          >
                            <h6
                              class="MuiTypography-root MuiTypography-h6 MuiTypography-noWrap css-z9ruai-MuiTypography-root"
                            />
                            <div
                              class="MuiStack-root css-kdc3n8-MuiStack-root"
                            >
                              <p
                                class="MuiTypography-root MuiTypography-body2 css-c2tnzq-MuiTypography-root"
                              >
                                Bookmark 1 description
                              </p>
                              <p
                                class="MuiTypography-root MuiTypography-body2 MuiTypography-noWrap css-176gd80-MuiTypography-root"
                              />
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                    <div
                      class="MuiBox-root css-k008qs"
                    >
                      <div
                        class="MuiBox-root css-rmtq2p"
                      >
                        <div
                          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-11rpxa7-MuiPaper-root"
                        >
                          <span
                            class="component-image MuiBox-root css-1jl2tef"
                            style="border-radius: 10px;"
                          >
                            <span
                              class="component-image-wrapper lazy-load-image-background blur"
                              style="background-image: url(/assets/placeholder.svg); background-size: 100% 100%; color: transparent; display: inline-block;"
                            >
                              <img
                                class="MuiBox-root css-19t2nho"
                                src=""
                              />
                            </span>
                          </span>
                          <div
                            class="MuiBox-root css-1ue9rvn"
                          >
                            <div
                              class="MuiBox-root css-1lekzkb"
                            >
                              <div
                                class="MuiStack-root css-1nnjrde-MuiStack-root"
                              >
                                <svg
                                  class="tabler-icon tabler-icon-heart-filled "
                                  fill="#F2F3F5"
                                  height="16"
                                  stroke="none"
                                  style="margin-bottom: 2px;"
                                  viewBox="0 0 24 24"
                                  width="16"
                                  xmlns="http://www.w3.org/2000/svg"
                                >
                                  <path
                                    d="M6.979 3.074a6 6 0 0 1 4.988 1.425l.037 .033l.034 -.03a6 6 0 0 1 4.733 -1.44l.246 .036a6 6 0 0 1 3.364 10.008l-.18 .185l-.048 .041l-7.45 7.379a1 1 0 0 1 -1.313 .082l-.094 -.082l-7.493 -7.422a6 6 0 0 1 3.176 -10.215z"
                                  />
                                </svg>
                                <p
                                  class="MuiTypography-root MuiTypography-body2 css-1571v2w-MuiTypography-root"
                                >
                                  200
                                </p>
                              </div>
                            </div>
                          </div>
                          <div
                            class="MuiCardContent-root css-gbn0xv-MuiCardContent-root"
                          >
                            <h6
                              class="MuiTypography-root MuiTypography-h6 MuiTypography-noWrap css-z9ruai-MuiTypography-root"
                            />
                            <div
                              class="MuiStack-root css-kdc3n8-MuiStack-root"
                            >
                              <p
                                class="MuiTypography-root MuiTypography-body2 css-c2tnzq-MuiTypography-root"
                              >
                                Bookmark 2 description
                              </p>
                              <p
                                class="MuiTypography-root MuiTypography-body2 MuiTypography-noWrap css-176gd80-MuiTypography-root"
                              />
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
